code/game/objects/structures/mirror.dm

```diff
@@ -156,6 +156,16 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror/broken, 28)
 		return TRUE
 
 	var/datum/species/newrace = selectable_races[racechoice]
+
+	newrace = new newrace()
+
+	var/attributes_desc = newrace.get_physical_attributes()
+
+	var/answer = tgui_alert(race_changer, attributes_desc, "Become a [newrace]?", list("Yes", "No"))
+	if(answer == "No")
+		change_race(race_changer) // try again
+		return
```

Comment:
```suggestion
	if(answer != "Yes")
		change_race(race_changer) // try again
		return
```

We MUST respect all user prompts, because if they close out of the tgui alert it will return `null` and the code will progress as if they had picked "Yes" in your current framework.